### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by yingbull**
### **Description**
- This PR introduces a comprehensive suite of unit tests for the healthcare domain managers, specifically focusing on the `ConsultationManagerImpl` and `ScheduleManagerImpl`.
- It enhances the test infrastructure by adding necessary DAOs and configurations to facilitate effective testing.
- The changes ensure that various scenarios, including CRUD operations, privilege checks, and conflict management, are thoroughly tested.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>ConsultationManagerUnitTest.java</strong><dd><code>Comprehensive Unit Tests for Consultation Manager</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/test-modern/java/io/github/carlos_emr/carlos/managers/ConsultationManagerUnitTest.java
<li>Added unit tests for <code>ConsultationManagerImpl</code> covering CRUD operations, <br>search functionality, and security checks.<br> <li> Implemented mocking for various DAOs and services used in the tests.<br> <li> Included tests for handling privileges and ensuring correct behavior <br>under different scenarios.<br>

</details>

  </td>
  <td><a href="https://github.com/carlos-emr/carlos/pull/313/files#diff-f859acfa9a00b78e73faaad3b37b166590efc0743ccfc7facc33500ed472ad8c">+1589/-0</a></td>
</tr>

<tr>
  <td>
    <details>
      <summary><strong>ScheduleManagerUnitTest.java</strong><dd><code>Unit Tests for Schedule Manager Functionality</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/test-modern/java/io/github/carlos_emr/carlos/managers/ScheduleManagerUnitTest.java
<li>Added unit tests for <code>ScheduleManagerImpl</code> covering appointment <br>management, scheduling, and security checks.<br> <li> Implemented tests for handling various appointment scenarios including <br>conflicts and retrievals.<br> <li> Mocked necessary DAOs and services for effective testing.<br>

</details>

  </td>
  <td><a href="https://github.com/carlos-emr/carlos/pull/313/files#diff-051702609a70cbc44259d59eed20badc68fa91e8674b8dd5e64e6acb05cf3d74">+1354/-0</a></td>
</tr>

<tr>
  <td>
    <details>
      <summary><strong>OpenOUnitTestBase.java</strong><dd><code>Improved Test Base Class with Logging Mocking</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/test-modern/java/io/github/carlos_emr/carlos/test/unit/OpenOUnitTestBase.java
<li>Enhanced <code>OpenOUnitTestBase</code> to include mocking for <code>LogAction</code> to silence <br>audit logging during tests.<br> <li> Added utility methods for dependency injection and mock management.<br>

</details>

  </td>
  <td><a href="https://github.com/carlos-emr/carlos/pull/313/files#diff-b9ffa4c43a9d23240fa9e397be1fb7aa68a275b1b2753b2065426a4cdf4c963e">+49/-2</a>&nbsp; &nbsp; </td>
</tr>
</table></td></tr><tr><td><strong>Configuration changes
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>test-context-full.xml</strong><dd><code>Updated Spring Context for Healthcare Testing</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/test-modern/resources/test-context-full.xml
<li>Updated Spring context configuration to include new DAOs for <br>healthcare domain entities.<br> <li> Added beans for <code>AllergyDao</code>, <code>AppointmentDao</code>, and other related DAOs for <br>testing purposes.<br>

</details>

  </td>
  <td><a href="https://github.com/carlos-emr/carlos/pull/313/files#diff-9ce008539620429d7d45b1b3de451bea61eed134e74e448f76ecb0f36d7a5e51">+70/-0</a>&nbsp; &nbsp; </td>
</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **Penify usage**:
>Comment `/help` on the PR to get a list of all available Penify tools and their descriptions

## Summary by CodeRabbit

* **Tests**
  * Added many comprehensive unit test suites and shared test bases for managers (appointments, consultations, labs, measurements, prescriptions, preventions, schedules, notes, allergies, security, tickler) improving coverage for privileges, consent, edge cases, and DAO interactions.
* **Chores**
  * Expanded test persistence/context entries and standardized static-mocking and dependency injection patterns to reduce test pollution and streamline setup/teardown.
* **Documentation**
  * Updated test naming guidelines to a clearer BDD-style convention with examples.

### CODE REVIEW

File: src/test-modern/java/io/github/carlos_emr/carlos/managers/AllergyManagerUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,909 @@
...
+
+            // Then
+            assertThat(result).isEmpty();
+        }
+
+        @Test
+        @DisplayName("should pass exact demographic ID and date to DAO")
+        void shouldPassExactParameters_whenPatientConsented() {
+            // Given
+            Integer specificDemo = 54321;
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/test-modern/java/io/github/carlos_emr/carlos/managers/AllergyManagerUnitTest.java]: fixed

**Comment by yingbull** [comment]
@coderabbitai

**Comment by yingbull** [comment]
@claude review all comments from from contributors and bots on this PR and determine which are correct, and recommend fixes.

**Comment by yingbull** [comment]
@claude fix 1-3

**Comment by yingbull** [comment]
@coderabbitai review

File: src/test-modern/java/io/github/carlos_emr/carlos/managers/ConsultationManagerUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,1589 @@
...
+            ConsultationRequest existingRequest = createRequestWithId(TEST_REQUEST_ID);
+            List<ConsultationRequestExt> extras = new ArrayList<>();
+            ConsultationRequestExt ext = createExt(null, TEST_REQUEST_ID, "appointmentYear", "2026");
+            extras.add(ext);
+            existingRequest.setExtras(extras);
+
+            // Stub the ext DAO calls
+            when(mockConsultationRequestExtDao.getConsultationRequestExts(TEST_REQUEST_ID))
+                .thenReturn(Collections.emptyList())
+                .thenReturn(extras);
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/test-modern/java/io/github/carlos_emr/carlos/managers/ConsultationManagerUnitTest.java:560]: closing, refactored, old review.

File: src/test-modern/java/io/github/carlos_emr/carlos/managers/ConsultationManagerUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,1589 @@
...
+            when(mockSecurityInfoManager.hasPrivilege(any(), eq("_con"), eq("r"), any()))
+                .thenReturn(false);
+
+            // When / Then
+            assertThatThrownBy(() -> consultationManager.search(mockLoggedInInfo, filter))
+                .isInstanceOf(RuntimeException.class)
+                .hasMessageContaining("Access Denied");
+        }
+    }
+
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/test-modern/java/io/github/carlos_emr/carlos/managers/ConsultationManagerUnitTest.java:733]: closing, refactored, old review.

File: src/test-modern/java/io/github/carlos_emr/carlos/managers/MeasurementManagerUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,1196 @@
...
+            String propKey = "mgroup.ds.html.999";
+            when(mockPropertyDao.checkByName(propKey)).thenReturn(null);
+
+            // When
+            measurementManager.removeMeasurementGroupDS(propKey);
+
+            // Then
+            verify(mockPropertyDao, never()).remove(anyInt());
+        }
+    }
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/test-modern/java/io/github/carlos_emr/carlos/managers/MeasurementManagerUnitTest.java:1115]: closing, refactored, old review.

**Comment by yingbull** [comment]
@coderabbitai review

File: src/test-modern/java/io/github/carlos_emr/carlos/managers/ConsultationManagerUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,1589 @@
...
+        void shouldReturnResponseCount_fromDao() {
+            // Given
+            ConsultationResponseSearchFilter filter = new ConsultationResponseSearchFilter();
+            when(mockConsultResponseDao.getConsultationCount(filter)).thenReturn(17);
+
+            // When
+            int count = consultationManager.getConsultationCount(filter);
+
+            // Then
+            assertThat(count).isEqualTo(17);
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/test-modern/java/io/github/carlos_emr/carlos/managers/ConsultationManagerUnitTest.java:768]: correcting.

**Comment by yingbull** [comment]
@claude review this PR and code currently and advise for any issues that need to be corrected.  Also, review the comments on this PR including from bots, assess if they are valid, and advise on corrections related to them.

**Comment by yingbull** [comment]
@claude determine if the adding of @tag to the nested classes is an issue and still needs to be done as per copilot, and if so, do this fix and commit to branch

File: src/test-modern/java/io/github/carlos_emr/carlos/managers/ScheduleManagerUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,1370 @@
...
+    // getAppointmentsForPatient
+    // -----------------------------------------------------------------------
+
+    /**
+     * Tests for {@link ScheduleManagerImpl#getAppointmentsForPatient}
+     * covering pagination and empty result sets.
+     */
+    @Nested
+    @DisplayName("getAppointmentsForPatient")
+    class GetAppointmentsForPatient {
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/test-modern/java/io/github/carlos_emr/carlos/managers/ScheduleManagerUnitTest.java:377]: should be fixed in latest.

**Comment by yingbull** [comment]
@claude fix the bare/try catch issue coderabbit identified

File: src/test-modern/java/io/github/carlos_emr/carlos/managers/SecurityInfoManagerUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,832 @@
...
+        @Test
+        @DisplayName("should define DELETE constant as 'd'")
+        void shouldDefineDeleteConstant() {
+            assertThat(SecurityInfoManager.DELETE).isEqualTo("d");
+        }
+
+        @Test
+        @DisplayName("should define NORIGHTS constant as 'o'")
+        void shouldDefineNoRightsConstant() {
+            assertThat(SecurityInfoManager.NORIGHTS).isEqualTo("o");
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/test-modern/java/io/github/carlos_emr/carlos/managers/SecurityInfoManagerUnitTest.java:228]: should be fixed in latest

File: src/test-modern/java/io/github/carlos_emr/carlos/managers/ScheduleManagerUnitTest.java [hunk 8]
```diff
@@ -0,0 +1,1365 @@
...
+                .thenReturn(expected);
+
+            // When
+            List<Appointment> result = scheduleManager.getDayAppointments(mockLoggedInInfo, TEST_PROVIDER, cal);
+
+            // Then
+            assertThat(result).hasSize(1);
+            verify(mockOscarAppointmentDao).findByProviderAndDayandNotStatus(
+                eq(TEST_PROVIDER), any(Date.class), eq(AppointmentStatus.APPOINTMENT_STATUS_CANCELLED));
+        }
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/test-modern/java/io/github/carlos_emr/carlos/managers/ScheduleManagerUnitTest.java:238]: should be fixed in latest

File: src/test-modern/java/io/github/carlos_emr/carlos/managers/SecurityInfoManagerUnitTest.java [hunk 9]
```diff
@@ -0,0 +1,832 @@
...
+
+    // =========================================================================
+    // Interface Constants
+    // =========================================================================
+
+    /** Tests that the interface constants defining privilege levels are correct. */
+    @Nested
+    @DisplayName("Interface Constants")
+    class InterfaceConstants {
+
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/test-modern/java/io/github/carlos_emr/carlos/managers/SecurityInfoManagerUnitTest.java:200]: should be fixed in latest

File: src/test-modern/java/io/github/carlos_emr/carlos/managers/SecurityInfoManagerUnitTest.java [hunk 10]
```diff
@@ -0,0 +1,839 @@
...
+                    }
+                    return Collections.emptyList();
+                });
+
+                boolean result = securityInfoManager.hasPrivilege(
+                    mockLoggedInInfo, TEST_OBJECT_NAME, SecurityInfoManager.READ,
+                    String.valueOf(TEST_DEMOGRAPHIC_NO));
+
+                assertThat(result).isFalse();
+            }
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/test-modern/java/io/github/carlos_emr/carlos/managers/SecurityInfoManagerUnitTest.java]: @claude we should fix this so it is as valid to code behavior in the application as possible with regard to test alignment

File: src/test-modern/java/io/github/carlos_emr/carlos/managers/PrescriptionManagerUnitTest.java [hunk 11]
```diff
@@ -0,0 +1,1011 @@
...
+            // When
+            try {
+                prescriptionManager.createNewPrescription(mockLoggedInInfo, drugs, TEST_DEMO_NO);
+            } catch (AccessDeniedException e) {
+                // Expected
+            }
+
+            // Then
+            verify(mockSecurityInfoManager).hasPrivilege(mockLoggedInInfo, "_rx", "w", TEST_DEMO_NO);
+        }
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/test-modern/java/io/github/carlos_emr/carlos/managers/PrescriptionManagerUnitTest.java:636]: @claude make sure this is fixed.

**Comment by yingbull** [comment]
@coderabbitai review

**Comment by yingbull** [comment]
@gemini-code-assist review

**Comment by yingbull** [comment]
@copilot review the PR and feedback and advise

**Comment by yingbull** [comment]
@claude review PR feedback and advise.

**Comment by yingbull** [comment]
@claude fix the 7 minor blocking issues and commit to this branch

**Comment by yingbull** [comment]
@claude review

File: src/test-modern/resources/test-lookup-tables.sql [hunk 12]
```diff
@@ -48,4 +48,28 @@ CREATE TABLE IF NOT EXISTS joint_admissions (
...
+INSERT INTO appointmentStatus (status, description, active, editable, icon, color)
+SELECT 'H', 'Here', 1, 1, '', '' WHERE NOT EXISTS (SELECT 1 FROM appointmentStatus WHERE status = 'H');
+
+-- Schedule template codes reference data
+INSERT INTO scheduletemplatecode (id, code, description, duration, color, confirm, bookinglimit)
+SELECT 1, 'A', 'Available', '15', '00FF00', '', 1 WHERE NOT EXISTS (SELECT 1 FROM scheduletemplatecode WHERE code = 'A');
+INSERT INTO scheduletemplatecode (id, code, description, duration, color, confirm, bookinglimit)
+SELECT 2, 'P', 'Primary Care', '15', '0000FF', '', 1 WHERE NOT EXISTS (SELECT 1 FROM scheduletemplatecode WHERE code = 'P');
+INSERT INTO scheduletemplatecode (id, code, description, duration, color, confirm, bookinglimit)
+SELECT 3, '-', 'Not Available', '15', 'FF0000', '', 0 WHERE NOT EXISTS (SELECT 1 FROM scheduletemplatecode WHERE code = '-');
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/test-modern/resources/test-lookup-tables.sql:75]: addressed in current
